testcases/kernel/syscalls/open/open15.c

```diff
@@ -0,0 +1,67 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later
+ * Copyright (C) 2020 Invisible Things Lab
+ *                    Michał Kowalczyk <mkow@invisiblethingslab.com>
+ */
+
+/*
+ * DESCRIPTION
+ *  Creates a file using `open` with bits outside of 07777 in `mode` set and
+ *  verifies if they were ignored.
+ *
+ * WARNING
+ *  The fact that these bits are ignored is not documented (at the time of
+ *  writing). Failure of this test doesn't necessarily mean that a regression
+ *  in Linux was introduced, its intention is to catch accidental interface
+ *  changes and warn kernel developers if that happens.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "tst_test.h"
+
+#define TEST_FILE "testfile"
+
+static int fd;
+
+static struct tcase {
+	char *filename;
+	int flags;
+	mode_t mode;
+} tcases[] = {
+	{TEST_FILE, O_RDWR | O_CREAT, 0644 | ~07777},
+	{TEST_FILE, 0, ~07777}
+};
+
+static void verify_open(unsigned int n)
+{
+	struct tcase *tc = &tcases[n];
+	struct stat buf;
+
+	TEST(open(tc->filename, tc->flags, tc->mode));
+	fd = TST_RET;
+	if (fd == -1) {
+		tst_res(TFAIL, "Cannot open the file");
```

Comment:
Two minor things: we use `TTERRNO` to print `errno` on failure. And we usually prefer to use return than have else block (readability).
```
if (fd == -1) {
	tst_res(TFAIL | TTERRNO, "Cannot open the file");
	return;
}

tst_res(TPASS, "Unknown mode bits were ignored as expected");
SAFE_CLOSE(fd);
```